src/tool_writeout.c

```diff
@@ -105,14 +107,84 @@ static const struct variable replacements[]={
   {"scheme", VAR_SCHEME},
   {"stdout", VAR_STDOUT},
   {"stderr", VAR_STDERR},
+  {"content", VAR_CONTENT},
+  {"error", VAR_ERROR},
   {NULL, VAR_NONE}
 };
 
+void copyandshift(char **replace, char *escape)
+{
+  int offset;
+  if(replace == NULL || escape == NULL) {
+    return;
+  }
+  offset = strlen(escape);
+  memcpy(*replace, escape, offset);
+  *replace += offset;
+}
+
+char *escape_string(char *output)
+{
+  char *formatStr = NULL, *tmpbuf = output,
+    *replaceStr = NULL, *escapeStr = NULL;
+  int len = 0, offset = 0;
+  if(NULL == output) {
+    return NULL;
+  }
+  len = strlen(output);
+  formatStr = (char *)malloc(2*len + 1);
+  memset(formatStr, 0, 2*len + 1);
+  replaceStr = formatStr;
+  while (*tmpbuf) {
+    switch(*tmpbuf) {
+    case '"':
+      escapeStr = "\\\"";
+      copyandshift(&replaceStr, escapeStr);
+      break;
+    case '\\':
+      escapeStr = "\\\\";
+      copyandshift(&replaceStr, escapeStr);
+      break;
+    case '\b':
+      escapeStr = "\\b";
+      copyandshift(&replaceStr, escapeStr);
+      break;
+    case '\f':
+      escapeStr = "\\f";
+      copyandshift(&replaceStr, escapeStr);
+      break;
+    case '\n':
+      escapeStr = "\\n";
+      copyandshift(&replaceStr, escapeStr);
+      break;
+    case '\r':
+      escapeStr = "\\r";
+      copyandshift(&replaceStr, escapeStr);
+      break;
+    case '\t':
+      escapeStr = "\\t";
+      copyandshift(&replaceStr, escapeStr);
+      break;
+    default:
+      if('\x00' <= *tmpbuf && *tmpbuf <= '\x1f') {
+        curlx_msprintf(replaceStr, "\\u%04x", (int)*tmpbuf);
```

Comment:
So this outputs `\u00bb` for byte values below 32 and I suppose over 127 too since tmpbuf points to a signed char? What made you pick that escaping? (and it's not documented either)

Reply:
For the shortest representation you may use JSON shortcuts, such as \" instead of \u0022. The function produces the shortest JSON representation of a UTF-8 encoded string

Reply:
1. JSON style is not necessarily the only way to format such bytes, so this needs to be documented.
2. There will me many cases where the data is not UTF-8.
3. Dereferencing the char pointer makes it a byte, not necessarily a full UTF-8 character surely?

Reply:
@bagder I will documented for it.  This escape format is only for JSON format.